lib/interceptor.js

```diff
@@ -154,52 +167,120 @@ var interceptor = {
       return headers || {};
     }
 
+    /**
+     * Stringify the given XHR payload so it can be parsed as JSON
+     * @param {*} payload XHR request body that is sent to the remote server.
+     * @returns {string} JSON-parsable representation of the request body
+     */
     function parsePayload(payload) {
-      var parsed;
       if (typeof payload == 'string') {
-        parsed = payload;
-      } else if (payload instanceof FormData) {
-        parsed = {};
+        return payload;
+      }
+      if (payload instanceof FormData) {
+        var parsed = {};
         var entries = payload.entries();
         var item;
         while (((item = entries.next()), !item.done)) {
           parsed[item.value[0]] = item.value.slice(1);
         }
-        parsed = JSON.stringify(parsed);
-      } else if (payload instanceof ArrayBuffer) {
-        parsed = String.fromCharCode.apply(null, payload);
-      } else {
-        // Just try to convert it to a string, whatever it might be
-        try {
-          parsed = JSON.stringify(payload);
-        } catch (e) {
-          parsed = '';
-        }
+        return JSON.stringify(parsed);
       }
-      return parsed;
+      if (payload instanceof ArrayBuffer) {
+        return String.fromCharCode.apply(null, payload);
+      }
+
+      // Just try to convert it to a string, whatever it might be
+      try {
+        return JSON.stringify(payload);
+      } catch (e) {
+        console.error(
+          '[wdio-intercept-service]: Failed to stringify payload as JSON!',
```

Comment:
I really like the error prefix here! Do you think it would be a good idea to put this in a constant somewhere?

Reply:
Done